moto/sagemaker/models.py

```diff
@@ -1105,6 +1105,12 @@ def gen_response_object(self) -> Dict[str, Any]:
             "AdditionalInferenceSpecifications",
             "SkipModelValidation",
         ]
+        if (
+            "ModelPackageGroupName" in response_object
+            and response_object["ModelPackageGroupName"]
+            == response_object["ModelPackageName"]
+        ):
+            del response_object["ModelPackageName"]
```

Comment:
I am wondering if it makes more sense to use the `model_package_type` for this, but we currently do not hold it. I am dropping a new commit with this change that we can later drop if that is the case.

Reply:
I've just dropped this on commit 6915d300cad6c01ed4d9a7d09ee047f208b42924.